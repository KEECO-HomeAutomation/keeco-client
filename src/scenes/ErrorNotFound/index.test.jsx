import React from 'react';
import { shallow } from 'enzyme';

import { ErrorNotFound } from './index';

import Button from '@material-ui/core/Button';

describe('<ErrorNotFound />', () => {
	var mockedHistory = null;
	var props = null;
	beforeEach(() => {
		mockedHistory = {
			push: jest.fn()
		};
		props = {
			classes: {
				centerPanel: ''
			},
			history: mockedHistory
		};
	});
	afterEach(() => {
		mockedHistory = null;
		props = null;
	});

	test('Should render without errors', () => {
		const comp = shallow(<ErrorNotFound {...props} />);
		expect(comp.length).toBe(1);
	});

	test('Should match snapshot', () => {
		expect(shallow(<ErrorNotFound {...props} />)).toMatchSnapshot();
	});

	test('Clicking on go to home button should redirect to /', () => {
		const comp = shallow(<ErrorNotFound {...props} />);
		comp.find(Button).simulate('click');
		expect(mockedHistory.push).toBeCalledTimes(1);
		expect(mockedHistory.push).toBeCalledWith('/');
	});
});
