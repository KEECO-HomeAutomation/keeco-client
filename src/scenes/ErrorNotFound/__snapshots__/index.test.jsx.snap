// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorNotFound /> Should match snapshot 1`] = `
<WithStyles(Paper)
  className=""
>
  <WithStyles(Typography)
    gutterBottom={true}
    variant="h6"
  >
    Error 404
  </WithStyles(Typography)>
  <WithStyles(Typography)>
    The requested page is not available. Sorry for that.
  </WithStyles(Typography)>
  <WithStyles(Button)
    color="primary"
  >
    Go to home!
  </WithStyles(Button)>
</WithStyles(Paper)>
`;
